elastic/kibana
# [Console] Toward a more reactive data flow model

## Summary

This PR is a continuation of https://github.com/elastic/kibana/pull/43346

It specifically focusses on refactoring items that were not included in the initial migration to NP Ready. Specifically:

- `app.js` and legacy initialisation
- `input.js` initialisation logic
- `output.js` initialisation logic

A lot of the logic for managing editor state was still in the quarantined section of the code base. This PR proposes an alternate model where we use React's context to drive UI updates.

## How to review

There are two primary ways that this PR should be reviewed:
1. Does it break any existing user functionality
    - Check that your settings are all still in place
    - Check that your history is still intact
    - Any other UI/UX anomalies around sending requests and UI components not updating
2. Read through the refactor
    - A lot of the driving logic is still considered legacy but now Editor UI updates are driven through a dispatch model
    - Review what is being done in `history.ts` service (i.e., is it sensible to opt out of the React model for this more special case or is there another solution?)
    - Editor instances have specifically been moved into a new stateful component called `EditorRegistry`. This is a stateful singleton where editor instances are stored outside of their respective components. The thinking here is that ace specific APIs should be kept in legacy.
    - Editor state (i.e., text values) are not stored inside of context state. The thinking here is that we would trigger to many unnecessary re-renders. Is this the correct path?
